contracts/sol6/reserves/bridgeReserve/uniswap/KyberUniswapReserve.sol

```diff
@@ -0,0 +1,371 @@
+pragma solidity 0.6.6;
+
+import "../../../IKyberReserve.sol";
+import "../../../IERC20.sol";
+import "../../../utils/WithdrawableNoModifiers.sol";
+import "../../../utils/Utils5.sol";
+import "../../../utils/zeppelin/SafeERC20.sol";
+
+interface IUniswapV2Router01 {
+
+    function swapExactETHForTokens(
+        uint256 amountOutMin,
+        address[] calldata path,
+        address to,
+        uint256 deadline
+    ) external payable returns (uint256[] memory amounts);
+
+    function swapExactTokensForETH(
+        uint256 amountIn,
+        uint256 amountOutMin,
+        address[] calldata path,
+        address to,
+        uint256 deadline
+    ) external returns (uint256[] memory amounts);
+
+    function getAmountsOut(uint256 amountIn, address[] calldata path)
+        external
+        view
+        returns (uint256[] memory amounts);
+
+    function getAmountsIn(uint256 amountOut, address[] calldata path)
+        external
+        view
+        returns (uint256[] memory amounts);
+
+    function factory() external pure returns (address);
+}
+
+interface IUniswapFactory {
+    function getPair(address tokenA, address tokenB) external view returns (address pair);
+}
+
+interface IUniswapV2Pair {
+    function getReserves()
+        external
+        view
+        returns (
+            uint112 reserve0,
+            uint112 reserve1,
+            uint32 blockTimestampLast
+        );
+}
+
```

Comment:
please separate interfaces to a different file.
can have it in same directory

Reply:
as later discuss, these interfaces are import from library